### ISSUE TITLE
Incorrect ref analysis diagnostic when assigning a Span to a readonly property

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#81796]
There should be no diagnostic in `ReadOnlyRefStruct_RefLikeProperty`, `ReadOnlyRefLikeProperty_01`, and `ReadOnlyRefLikeProperty_02`, because the setter is readonly (by virtue of the containing struct being readonly, the containing property being readonly, or the setter being readonly).

To illustrate the scenario, here's the code for last test:
```
using System;

public ref struct S<T>
{
    public Span<T> P { get => default; readonly set {} }

    public unsafe static void N(S<byte> b)
    {
        Span<byte> x = stackalloc byte[5];
        b.P = x; // incorrect diagnostic: warning CS9080: Use of variable 'x' in this context may expose referenced variables outside of their declaration scope
    }
}
```
